tensorflow/lite/micro/tools/generate_test_for_model.py

```diff
@@ -0,0 +1,223 @@
+                          dtype=dtype,
+                          size=input_tensor0().shape),
+        np.random.randint(low=np.iinfo(dtype).min,
+                          high=np.iinfo(dtype).max,
+                          dtype=dtype,
+                          size=input_tensor1().shape)
+    ]
+
+  def _generate_inputs_transpose_conv(self, interpreter, dtype):
+    input_tensor0 = interpreter.tensor(0)
+    filter_tensor = interpreter.tensor(1)
+    input_tensor1 = interpreter.tensor(2)
+    output_shape = np.array([
+        1,
+        filter_tensor().shape[1] + input_tensor1().shape[1] - 1,
+        int(input_tensor1().shape[2] * 2 + filter_tensor().shape[2] / 2),
+        filter_tensor().shape[0]
+    ],
+                            dtype=np.int32)
+    return [
+        output_shape,
+        np.random.randint(low=np.iinfo(dtype).min,
+                          high=np.iinfo(dtype).max,
+                          dtype=dtype,
+                          size=input_tensor1().shape)
+    ]
+
+  def _GetTypeStringFromTensor(self, tensor):
+    if tensor.dtype == np.int8:
+      return 'int8'
+    if tensor.dtype == np.int16:
+      return 'int16'
+    if tensor.dtype == np.int32:
+      return 'int32'
+    if tensor.dtype == np.float:
+      return 'float'
+
+  def generate_golden_single_in_single_out(self):
+    """ Takes a single model as input. It is expecting a list with one model.
+        It then generates input and output in CSV format for that model. """
+
+    if (len(self.model_paths) != 1):
+      raise RuntimeError(f'Single model expected')
+    model_path = self.model_paths[0]
+    interpreter = tf.lite.Interpreter(model_path=model_path)
+
+    interpreter.allocate_tensors()
+
+    input_details = interpreter.get_input_details()
+    if len(input_details) > 1:
+      raise RuntimeError(f'Only models with one input supported')
+    input_tensor = interpreter.tensor(
+        interpreter.get_input_details()[0]['index'])
+    output_tensor = interpreter.tensor(
+        interpreter.get_output_details()[0]['index'])
+
+    input_type = interpreter.get_input_details()[0]['dtype']
+    output_type = interpreter.get_output_details()[0]['dtype']
+    if input_type != np.int8 or output_type != np.int8:
+      raise RuntimeError(f'Only int8 models supported')
+
+    generated_inputs = self._generate_inputs_single(interpreter,
+                                                    input_tensor().dtype)
+    for i, _input_detail in enumerate(input_details):
+      interpreter.set_tensor(input_details[i]["index"], generated_inputs[i])
+
+    interpreter.invoke()
+
+    self._write_golden(generated_inputs, model_path, output_tensor)
+
+  def generate_goldens(self, builtin_operator):
+    """ Takes a list of one or more models as input.
+        It also takes a built in operator as input because the generated input depends
+        on what type of operator it is, and it supports a limited number of operators.
```

Comment:
Should we add the following  as well:  "All models in the list assumes the operator as the first operator."

Reply:
Done